### ISSUE TITLE
Confirmation to send Active Note to use as context

### ISSUE THREAD

**Issue by o-mikhailovskii** [logancyang/obsidian-copilot#696]
Sometimes, notes might contain sensitive information, and one would rather not accidentally send them to an LLM provider. It would be nice to have an option in the settings to ask for confirmation before sending an Active Note to the prompt. Couldn't find one in the current setup.

**Comment by logancyang** [comment]
Makes sense. It can be quite disruptive if every note has to go through this though. Is it better to use tags and let copilot ask for confirmation only for the tagged notes?

**Comment by o-mikhailovskii** [comment]
I do not often use the Send button, but you are right. Building upon your suggestion, one could implement a checkbox (to ask for confirmation) paired with a text (or any other suitable) field of tags. This field would allow the user to select only from the tags available in the vault. If the field is empty, the rule applies to all notes. One has to bear in mind that there might be too many tags to pre-populate this field and let the user "turn on" the desired tags.

**Comment by o-mikhailovskii** [comment]
I was primarily talking exactly about "Send Note(s) to Prompt" button, so yes, its deprecation would resolve the issue.

Speaking of Vault QA mode, I see that you could exclude specific folders and notes. Tags are somewhat like folders in terms of notes' organization but nont the same. So, would it be possible/easy to add them as an option as well? Idk, but does the "Refresh Index for Active Note" button respect these exclusions?

**Comment by logancyang** [comment]
Yes exclusion by tags should be supported for sure, I'm gonna add it next.

All indexing actions respect the exclusion rule. BUT if your exclusion rule changed, "Refresh Index" won't delete the already indexed stuff, so in that case you should rebuild the index with "Force re-index" command (or "Clear the vector store" and reindex).

**Comment by o-mikhailovskii** [comment]
Got it! Thanks for the clarification. Will be waiting for the exclusion by tag feature. As well as the deprecation of the "Send Note(s) to Prompt" button 🙂

**Comment by logancyang** [comment]
Turns out excalidraw files are still markdown `.excalidraw.md` lol, so excluding by folder should be enough for now.
